### ISSUE TITLE
BBB 2.3 not rendering the presentation

### ISSUE THREAD

**Issue by lonesomewalker** [tilmanmoser/bbb-video-download#55]
Did the setup as described to render ALL recordings after processing.
As in 2.2 in bbb-record the rb script is shown as post-processing, this is somewhat missing in 2.3.

Error log: empty :-/

Manual starting:
```
root@conference /opt/bbb-video-download # sudo -u bigbluebutton docker-compose run --rm --user 998:998 app node index.js -i /var/bigbluebutton/published/presentation/0c2d0e4a7907f4da5ca73ba8399a4be17a189b6d-1615618784968 -o /var/bigbluebutton/published/presentation/0c2d0e4a7907f4da5ca73ba8399a4be17a189b6d-1615618784968/video.mp4
Creating network "bbb-video-download_default" with the default driver
WARNING: Error loading config file: /root/.docker/config.json: open /root/.docker/config.json: permission denied
[Start] Rendering downloadable video for BBB presentation  {
  args: Namespace(input='/var/bigbluebutton/published/presentation/0c2d0e4a7907f4da5ca73ba8399a4be17a189b6d-1615618784968', output='/var/bigbluebutton/published/presentation/0c2d0e4a7907f4da5ca73ba8399a4be17a189b6d-1615618784968/video.mp4'),
  format: 'mp4',
  docker: true
}
```
This renders content to ./tmp

**Comment by tilmanmoser** [comment]
@ichdasich pointed out, that for bbb2.3 you need an additional parameter in the snippet calling the script. The snippet was updated, please update to the latest version and update the post_publish_hook according to the README:

```
# (as root or with sudo)
cd /opt/bbb-video-download
export BBB_VIDEO_DOWNLOAD_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
export BBB_UID="$(cat /etc/passwd | grep bigbluebutton | cut -d: -f3)"
export BBB_GID="$(cat /etc/passwd | grep bigbluebutton | cut -d: -f4)"

## bbb v2.3:
envsubst < ./snippets/bbb23_post_publish_bbb_video_download.rb.template > /usr/local/bigbluebutton/core/scripts/post_publish/a0_post_publish_bbb_video_download.rb
```

**Comment by lonesomewalker** [comment]
Not closed, see #64

**Comment by tilmanmoser** [comment]
Well, #64 is about using the wrong uid/gid. Deploying script as posted would not result in wrong uid/gid. So there must be something different going on. Unfortunately I have no v2.3 running yet. The v2.3 hook was a user submission. But I have user reporting that the script itself is still working with v2.3; its just the post publish hook that is causing trouble. It might take me some time to update. I‘d sppreciate any help on this topic.

**Comment by lonesomewalker** [comment]
Closed #64 because it is done from my side, and a simple addendum to the readme.md solves that for everybody :-)

Dunno why this doesn't work automatically, i totally agree that the post publish should work because the uid/gid is taken from /etc/passwd:
Well, since we have BBB2.3 stable now and they changed uid/gid... (what a nonsense... funny for Scalelite also...)

Will dig in this when there is a bit more spare time.
